### PR TITLE
Fix Authorization Failure While Reading Tables From Unity Catalog [databricks]

### LINKED ISSUE #10318: [BUG] `fs.azure.account.keyInvalid` configuration issue while reading from Unity Catalog Tables on Azure DB
**Describe the bug**
While using Azure Databricks and attempting to read a **Managed Table** from the Unity Catalog Metastore with the RAPIDS Accelerator, I encountered invalid credentials issue with the following message: `Failure to initialize configuration for storage account databricksmetaeast.dfs.core.windows.net: Invalid configuration value detected for fs.azure.account.keyInvalid configuration value detected for fs.azure.account.key.` However, this error doesn't occur when RAPIDS is disabled.

**Notes**
 Adding the credentials of the storage container to the Spark configuration properties can serve as an interim solution. However, this approach is not scalable when there are multiple storage containers.

**Environment details**
Managed Tables on Azure Databricks with Unity Catalog and RAPIDS Accelerator

### PR DESCRIPTION
**Pull request by razajafri**
This PR replaces ThreadPoolExecutor creation with a utility function from Spark. The idea behind this replacement is to benefit from the custom implementation in different flavors of Spark. 

We also reverted the changes made in [5f40711](https://github.com/NVIDIA/spark-rapids/commit/5f4071191ab5b60b67f12dcf92693b2e233b8ad9) because they were interfering with some confs that were set by Databricks runtime which are essential to run with Unity Catalog. 

fixes #10318

### CODE REVIEW

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala
Comment (jlowe) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala:1]: The spark311 version of this file also should be removed.

**Review by jlowe — COMMENTED** [review]
There are many other places in the code where threadpools are created, and those should also be updated to use Spark's ThreadUtils rather than roll their own.  Filed #10759 to track that separately, as it's not needed to fix the Unity catalog issue.

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala
Comment (razajafri) [reply sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala:1]: Done, there was a merge conflict which I thought I resolved by removing the file but looks like I made a mistake. PTAL

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**